#
# (Exim 4 wish list #163) suppress the "Received:" header if
# received_header_text is empty.
#
--- src/receive.c.orig	Mon May 12 15:39:21 2003
+++ src/receive.c	Fri Jun 13 03:17:26 2003
@@ -1915,9 +1915,17 @@
 /* The first element on the header chain is reserved for the Received
 header, so all we have to do is fill in the text pointer. */
 
-header_list->text = string_sprintf("%s; %s\n", received, timestamp);
+if (received[0] == 0)
+  {
+  header_list->text = string_sprintf("Received: ; %s\n", timestamp);
+  header_list->type = htype_old;
+  }
+else
+  {
+  header_list->text = string_sprintf("%s; %s\n", received, timestamp);
+  header_list->type = htype_received;
+  }
 header_list->slen = Ustrlen(header_list->text);
-header_list->type = htype_received;
 
 
 /* If there is no From: header, generate one. If there is no sender address,
